### PR TITLE
feat: report daily participants to public stats

### PR DESCRIPTION
**Pull request by bajtos**
Create a new `spark_stats` table to keep track of daily participants:

```sql
SELECT day::TEXT, COUNT(DISTINCT participant_id) as count
FROM daily_participants GROUP BY day;
```

This table allows us to correctly calculate monthly participants too:

```sql
SELECT
  date_trunc('month', day)::DATE::TEXT as month,
  COUNT(DISTINCT participant_id) as count
FROM daily_participants
GROUP BY month;
```

TODO:
- [x] add tests

Links:
- https://github.com/filecoin-station/spark-evaluate/issues/79

### CODE REVIEW

File: lib/public-stats.js [hunk 1]
```diff
@@ -28,10 +44,48 @@ export const updatePublicStats = async ({ createPgClient, honestMeasurements })
...
+  ])
+}
+
+/**
+ * @param {import('./typings').CreatePgClient} createPgClient
+ * @param {string[]} participants
+ */
+const updateDailyParticipants = async (pgClient, participants) => {
+  debug('Updating daily participants (%s seen)', participants.length)
+  for (const participantAddress of participants) {
```
Comment (juliangruber) [inline lib/public-stats.js:58]: Are we ok with performing 2 SQL queries for every participant, in series? If it fails midway, will we have inconsistent data? Playing devil's advocate here

**Review by juliangruber — APPROVED** [review]
..otherwise LGTM!

File: lib/public-stats.js (hunk 1 above)
Comment (bajtos) [reply lib/public-stats.js:58]: Great questions! 💯 

I'll think about this tomorrow.

Comment (bajtos) [reply lib/public-stats.js:58]: Yeah, I was focused so much on storage efficiency and the querying side, that I completely neglected the writing side.

With ~2k participants per round, my current implementation would run 4k SQL queries. That would take forever to complete.

Thanks for flagging this early! 😍

Comment (bajtos) [reply lib/public-stats.js]: Cool, I found a neat trick: We can use `SELECT UNNEST($1::TEXT[])` to build a query that accepts a JavaScript array parameter and converts the items into result rows.

Then I can change `INSERT...VALUES` to `INSERT...SELECT` to leverage this mechanism.

The changes were a bit more involved (see [7a2d14d](https://github.com/filecoin-station/spark-evaluate/pull/133/commits/7a2d14d11f10513e2946eba863f4721383ed322f)), but I expect the performance to be excellent.

**Review by juliangruber — APPROVED** [review]
Great work 😍

**Pull request merged**